[Table Visualization] Replace table visualization with React and DataGrid (#2863)
opensearch-project/OpenSearch-Dashboards: src/plugins/vis_type_table/README.md

# Data Table

This is an OpenSearch Dashboards plugin that is used to visualize data and aggregations in tabular format.

## Create Data Table
To create a data table in OpenSearch Dashboards, first select `Visualize` from the navigation menu. Then click `Create Visualization` and choose `Data Table` as the visualization type.

## Select Metrics

### Metrics Aggregation

> At the `Metrics`, select the metric aggregation type from the menu and config it accordinly.

Reviewer: Do we want to clarify that by default, as you add more metrics, they will be added as new columns?

Author: Yeah I could add some clarifications.

```
At the `Metrics`, select the metric aggregation type from the menu and configure it accordingly. You could also add multiple metrics and each metrics is a separate column in table visualization.
```

### Buckets Aggregation

> At the `Buckets`, config the columns to be displayed in the table visualization.

Reviewer: ```suggestion
At the `Buckets`, configure the columns to be displayed in the table visualization.
```

Author: will change

> - `Split Rows` is used when you want to divide one row into more categories. It splits one row into multiple and add columns based on the categories you choose to split. For example, if you split the data based on gender then you want to know more on each gender's clothing preference. You could click `Split Rows` and input `clothing.category` in terms. Each genger's data is now splitted to multiple rows based on a new added column `clothing.category`.

Reviewer: some rewording suggestions:
```suggestion
- `Split Rows` is used when you want to divide one row into more based on some category. It splits one row into multiple and add columns based on the categories you choose to split. For example, if you split the data based on gender then you want to know more on each gender's clothing preference. You could click `Split Rows` and input `clothing.category` in terms. Each gender's data is now split across to multiple rows based on a new added column `clothing.category`.
```

But I'm still not sure this is quite right. In general, choosing split rows doesn't generally add columns. It just applies your existing columns to whatever field you split by.

Author: Will use the suggested rewording.

For the second question, from my observation, it always add columns when we split rows. I think the meaning of `split rows` in terms aggregation is to split on row into multiple based on the new split option. 

For example, we could have split rows with term aggregation. We could count column is split to two based on day_of_week.
<img width="732" alt="Screen Shot 2022-12-19 at 16 15 57" src="https://user-images.githubusercontent.com/79961084/208552658-c08619cd-9030-4a82-a872-c9ec8c2a872d.png">

Continue, split row again with term aggregation of field category.keyword. Now each row is split into multiple based on diff category. A new `category.keyword` col is added.
<img width="731" alt="Screen Shot 2022-12-19 at 16 15 44" src="https://user-images.githubusercontent.com/79961084/208552749-ee7b83d7-112e-4ccb-8297-148ae8ecfbd4.png">

It is not just term aggregation. Rows can be split via other aggregations as well, like Date Histogram, Histogram and etc, as long as this bucket aggregation has a field. The only bucket aggregation which doesn't have a field is `filter`. With setting a filter, it will add a filter col.

* Filter bucket agg

<img width="1045" alt="Screen Shot 2022-12-19 at 16 28 14" src="https://user-images.githubusercontent.com/79961084/208553740-8b897124-d05e-4e1a-8316-cb4862b9f6e1.png">

* Range bucket agg
<img width="735" alt="Screen Shot 2022-12-19 at 16 25 37" src="https://user-images.githubusercontent.com/79961084/208553780-f2f4f3d2-057c-42e0-9218-c896bb70f9bb.png">

* Histogram
<img width="737" alt="Screen Shot 2022-12-19 at 16 24 37" src="https://user-images.githubusercontent.com/79961084/208553839-0ffab04a-5a16-431a-97cc-725ef51b431e.png">

So I think it should be okay to remind user that `split row` will split one row and add a column to the table.

Reviewer: You're right, that when splitting rows, we always need to add a column for identifying the value of the agg that we're splitting on.

> - `Split Table` splits the table into seperate tables for the aggregation you choose. It is similar to `Split Rows`, but this time you will see multiple tables.

Reviewer: Could add that the tables can be arranged horizontally or vertically, but can't be wrapped.

Author: ```
`Split Table` splits the table into separate tables for the aggregation you choose. It is similar to `Split Rows`, but this time each row becomes a single table with aggregation columns arranged horizontally or vertically .
```

## Select Options

> In the `Options` tab, you could config more options.

Reviewer: ```suggestion
In the `Options` tab, you can configure more options.
```

Author: will change

- `Max rows per page` is the maximum number of rows displayed per page.

> - `Show metrics for every bucket/level` adds metrics aggregation to every columns.

Reviewer: ```suggestion
- `Show metrics for every bucket/level` adds metrics aggregation to every column.
```
this could use a little more info - it's not an obvious one.

Author: will change

- `Show partial rows` will include data with missing columns.

> - `Show total` calculates the selected metrics per column and displays the result at the bottom.

Reviewer: ```suggestion
- `Show total` calculates the selected metrics per column and displays the result at the bottom. Warning - depending on your columns and the total aggregation function selected, this may generate statistically invalid results. For example, avoid summing or averaging averages.
```

Author: will change. Thanks for providing such detailed comments.

- `Percentage column` adds one percentage column based on the chosen metrics aggregation.

> ## Example

Reviewer: ❤️

Author: Finally, have a correct one~~ lolll
